questionnaires/models.py

```diff
@@ -663,16 +663,26 @@ def get_context(self, request, *args, **kwargs):
             form_class = self.get_form_class()
             form = form_class(data=request.POST, page=self, user=request.user)
 
-            for k, v in form.fields.items():
-                form.fields[k].widget.attrs['readonly'] = True
+            # for k, v in form.fields.items():
+            #     form.fields[k].widget.attrs['readonly'] = True
 
             fields_info = {}
 
             total = 0
             total_correct = 0
+            form_data = dict(form.data)
             for field in self.get_form_fields():
-                # TODO: handle multi-value case
-                is_correct = form.data.get(field.clean_name) == field.correct_answer
+                if field.field_type in VALID_SKIP_SELECTORS:
+                    correct_answer = field.correct_answer.split(',')
+                else:
+                    correct_answer = field.correct_answer
+
+                if field.field_type == 'checkbox':
+                    answer = form_data.get(field.clean_name, 'off')
+                else:
+                    answer = form_data.get(field.clean_name)
+
+                is_correct = answer == correct_answer
```

Comment:
create a set() and then compare the answers

Reply:
done